Lib/python/std_filesystem.i

```diff
@@ -0,0 +1,90 @@
+/*
+  filesystem::path
+*/
+
+%{
+#include <filesystem>
+#include <Python.h>
+%}
+
+%fragment("common", "header") {
+    PyObject * importPathCls() {
+        PyObject * module = PyImport_ImportModule("pathlib");
+        PyObject * cls = PyObject_GetAttrString(module, "Path");
+        
+        Py_DECREF(module);
+        
+        return cls;
+    }
+
+    bool isPathInstance(PyObject * obj) {
+        PyObject * cls = importPathCls();
+        bool is_instance =  PyObject_IsInstance(obj, cls);
+
+        Py_DECREF(cls);
+
+        return is_instance;
+    }
+
+    const char * pathToStr(PyObject * obj) {
+        PyObject * str_obj = PyObject_Str(obj);
+        const char * s = PyUnicode_AsUTF8(str_obj);
+
+        Py_DECREF(str_obj);
+
+        return s;
+    }
+}
+
+%typemap(in, fragment="common") std::filesystem::path {
+    if (PyUnicode_Check($input)) {
+        const char* s = PyUnicode_AsUTF8($input);
+        $1 = std::filesystem::path(s);
+    }
+    else if (isPathInstance($input)) {
+        const char * s = pathToStr($input);
+        $1 = std::filesystem::path(s);
```

Comment:
Are you sure that this call will decode UTF-8 on Windows? If I read the specs correct, this is only done for `char8_t`, so only for unsigned char.

Should there be checks in the test-suite which checks for unicode edge cases instead of just foo?